docs/dev/go-style-guide.md

```diff
@@ -0,0 +1,74 @@
+
```

Comment:
should we link this somewhere directly on the main README? It might be nice to highlight this somewhere with more visibility. Perhaps even a PR template?

Reply:
Will add a link to it in the PR template.